File: Easy/2778_Sum_of_Squares_of_Special_Elements/Solution.go
package main

import "fmt"

// Past the function from leetcode here
func sumOfSquares(nums []int) int {
	n := len(nums)
	ans := 0

	for i := 0; i < n; i++ {
		if n%(i+1) == 0 {
			ans += nums[i] * nums[i]
		}
	}

	return ans
}

func main() {
	// test cases 1
	nums1 := []int{1, 2, 3, 4}
	fmt.Println(sumOfSquares(nums1))

	// test cases 2
	nums2 := []int{2, 7, 1, 19, 18, 3}
	fmt.Println(sumOfSquares(nums2))

}

File: Easy/2778_Sum_of_Squares_of_Special_Elements/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    int sumOfSquares(vector<int>& nums) {
        int n = nums.size();
        int ans = 0;

        for (int i = 0; i < n; i++) {
            if (n % (i + 1) == 0) {
                ans += nums[i] * nums[i];
            }
        }
        
        return ans;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    vector<int> nums1 = {1,2,3,4};
    cout << solution.sumOfSquares(nums1) << endl;

    // test cases 2
    vector<int> nums2 = {2,7,1,19,18,3};
    cout << solution.sumOfSquares(nums2) << endl;

    return 0;
}